src/composables/incomingTransactions.ts

```diff
@@ -0,0 +1,56 @@
+import { roundAmountToPrecision } from '../popup/utils';
+import { i18n } from '../store/plugins/languages';
+import { IDefaultComposableOptions, ITransaction } from '../types';
+import { useAccounts } from './accounts';
+import { useNotifications } from './notifications';
+import { useGetter } from './vuex';
+import { useWebSocket } from './webSocket';
+
+export function useIncomingTransactions({ store }: IDefaultComposableOptions) {
+  const { channelSubscribe } = useWebSocket({ store });
+  const { addWalletNotification } = useNotifications({ store });
+  const { isLocalAccountAddress, getAccountNameByAddress } = useAccounts({ store });
+  const getTxAmountTotal = useGetter<any>('getTxAmountTotal');
+  const getTxSymbol = useGetter<any>('getTxSymbol');
+
+  /**
+   * This function starts listening for incoming transactions on the 'Transactions' channel.
+   */
+  function startListeningForIncomingTransactions() {
+    const receivedTransactions: Set<string> = new Set<string>();
+
+    channelSubscribe('Transactions', (transaction: ITransaction) => {
+      if (!transaction?.tx) return;
+      const { hash } = transaction;
+      const { senderId, recipientId } = transaction.tx;
+
+      if (
+        recipientId
+        && isLocalAccountAddress(recipientId)
+        && !receivedTransactions.has(hash)
+      ) {
+        receivedTransactions.add(hash);
+
+        const amount = roundAmountToPrecision(getTxAmountTotal.value(transaction));
+        const title = i18n.t('pages.notifications.incomingTransaction');
+        const text = `${getAccountNameByAddress(recipientId)} ${i18n.t('pages.transactions.received')} ${amount} ${getTxSymbol.value(transaction)}`;
+
+        addWalletNotification({
+          id: hash,
+          text,
+          title,
+          buttonLabel: i18n.t('pages.notifications.viewTransaction'),
+          path: `/account-details/transactions/${hash}`,
```

Comment:
I suggest to use route names instead of direct path. In this case to generate path (as far as I remember) you can use:
```
router.resolve({ name: ROUTE_TX_DETAILS, params: { hash } });
```